api/coreservice.go

```diff
@@ -1379,25 +1383,35 @@ func (core *coreService) LogsInRange(filter *logfilter.LogFilter, start, end, pa
 		return nil, err
 	}
 
-	// TODO: improve using goroutine
 	if paginationSize == 0 {
 		paginationSize = 1000
 	}
 	if paginationSize > 5000 {
 		paginationSize = 5000
 	}
+
+	c := make(chan []*iotextypes.Log)
 	logs := []*iotextypes.Log{}
+	if len(blockNumbers) == 0 {
+		return logs, nil
+	}
 	for _, i := range blockNumbers {
-		logsInBlock, err := core.LogsInBlock(filter, i)
```

Comment:
it is not encouraged to call a public function of the same struct inside of another function of the struct.

Reply:
I can replace `core.LogsInBlock` to `core.logsInBlock`